admin/server/projects.go

```diff
@@ -190,11 +190,15 @@ func (s *Server) GetProject(ctx context.Context, req *adminv1.GetProjectRequest)
 
 			spec := resp.Resource.GetExplore().State.ValidSpec
 			if spec != nil {
-				condition.WriteString(fmt.Sprintf(" OR '{{.self.kind}}'='%s' AND '{{lower .self.name}}'=%s", runtime.ResourceKindMetricsView, duckdbsql.EscapeStringValue(strings.ToLower(spec.MetricsView))))
+				condition.WriteString(fmt.Sprintf(" OR ('{{.self.kind}}'='%s' AND '{{lower .self.name}}'=%s)", runtime.ResourceKindMetricsView, duckdbsql.EscapeStringValue(strings.ToLower(spec.MetricsView))))
 			}
 		}
 
 		attr = mdl.Attributes
+		// add the magic token's creator id to the attributes useful for checks in security rules
+		if mdl.CreatedByUserID != nil && *mdl.CreatedByUserID != "" {
+			attr["creator_id"] = *mdl.CreatedByUserID
+		}
```

Comment:
Just commenting here to reflect the discussion in Slack: the problem with this approach is that it breaks the report if the creator of it leaves (`CreatedByUserID` is nullable), even though other admins can still manage the report.

Is there a way we can explicitly grant access to the magic token's resource? If you look at the rule added below, it is in fact a deny rule for anything that doesn't match the conditions here – which is slightly different.